QEfficient/transformers/cache_utils.py

```diff
@@ -0,0 +1,128 @@
+# -----------------------------------------------------------------------------
+#
+# Copyright (c)  2023-2024 Qualcomm Innovation Center, Inc. All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+#
+# -----------------------------------------------------------------------------
+
+
+from typing import Any, Dict, List, Optional, Tuple
+import torch
+from transformers.cache_utils import (
+    DynamicCache
+)
+
+
+class QEffDynamicCache(DynamicCache):
+    """
+    A cache that grows dynamically as more tokens are generated. This is the default for generative models.
+
+    It stores the Key and Value states as a list of tensors, one for each layer. The expected shape for each tensor is
+    `[batch_size, num_heads, seq_len, head_dim]`.
+    """
+
+    def __init__(self) -> None:
```

Comment:
section of source already present in base class DynamicCache, having the section from line:24 to line:53 defy the use of class inheritance